### ISSUE TITLE
encoder_dims does not exist in the metadata

### ISSUE THREAD

**Issue by kislay99** [k2-fsa/sherpa-onnx#1336]
I'm using this model: [sherpa-onnx-zipformer-large-en-2023-06-26.tar.bz2](https://github.com/k2-fsa/sherpa-onnx/releases/download/asr-models/sherpa-onnx-zipformer-large-en-2023-06-26.tar.bz2)

when building and testing on iPhone I see the following error:
```
/Users/kislaysingh/offile_tts/sherpa-onnx/sherpa-onnx/csrc/online-zipformer-transducer-model.cc:InitEncoder:98 encoder_dims does not exist in the metadata
```

I tried making model type both zipformer and zipformer2 getting the same error. Although I'm able to run the following successfully on the iPhone: sherpa-onnx-zipformer-en-2023-04-01

Basically this [hugging face playground](https://huggingface.co/spaces/k2-fsa/web-assembly-asr-sherpa-onnx-en) model does exactly what I'm looking for, I'm guessing it's the large one.

Do I need to use some other configs?

**Comment by csukuangfj** [comment]
An offline/non-streaming model cannot be used with an OnlineRecognizer.

The error log is expected.

Your usage is not correct.
